### PR TITLE
Go: More Views examples

### PR DESCRIPTION
**Pull request by pellared**
Per https://github.com/open-telemetry/opentelemetry.io/pull/3275#discussion_r1327263845

Fixes https://github.com/open-telemetry/opentelemetry-go/issues/3186

**Please double-check if there is anything missing in the Go metrics docs.
See: https://deploy-preview-3285--opentelemetry.netlify.app/docs/instrumentation/go/manual/#metrics**

## Docs PR Checklist

- [x] This PR is for a documentation page whose authoritative copy is in the
      opentelemetry.io repository.

### CODE REVIEW

**Review by hanyuancheung — APPROVED** [review]

File: content/en/docs/instrumentation/go/manual.md [hunk 1]
```diff
@@ -780,6 +780,64 @@ meterProvider := metric.NewMeterProvider(
...
+  },
+  metric.Stream{Aggregation: metric.AggregationDrop{}},
+)
+
+meterProvider := metric.NewMeterProvider(
+	metric.WithView(view),
+)
+```
+
+For example, here's how you create a view removes the `http.request.method`
```
Comment (dmathieu) [inline content/en/docs/instrumentation/go/manual.md:804]:
```suggestion
For example, here's how you create a view that removes the `http.request.method`
```

Comment (dmathieu) [inline content/en/docs/instrumentation/go/manual.md:804]: The `For example` here is a bit repetitive with the one above. Suggestion:

```suggestion
As a second example, here's how you create a view removes the `http.request.method`
```

File: content/en/docs/instrumentation/go/manual.md [hunk 2]
```diff
@@ -780,6 +780,64 @@ meterProvider := metric.NewMeterProvider(
 )
 ```
 
+The SDK filters metrics and attributes before exporting metrics. This allows
+views to be used to e.g. reduce memory usage of high cardinality metrics or drop
+attributes that can contain sensitive data.
```
Comment (theletterf) [inline content/en/docs/instrumentation/go/manual.md:785]:
```suggestion
The SDK filters metrics and attributes before exporting metrics. For example, you
can use views to reduce memory usage of high cardinality metrics or drop
attributes that might contain sensitive data.
```

File: content/en/docs/instrumentation/go/manual.md [hunk 3]
```diff
@@ -780,6 +780,64 @@ meterProvider := metric.NewMeterProvider(
...
+
+For example, here's how you create a view that drops the `latency` instrument
+from the `http` instrumentation library:
```
Comment (theletterf) [inline content/en/docs/instrumentation/go/manual.md:788]:
```suggestion
Here's how you create a view that drops the `latency` instrument
from the `http` instrumentation library:
```

File: content/en/docs/instrumentation/go/manual.md [hunk 4]
```diff
@@ -780,6 +780,64 @@ meterProvider := metric.NewMeterProvider(
...
+)
+
+meterProvider := metric.NewMeterProvider(
+	metric.WithView(view),
+)
+```
+
+For example, here's how you create a view that removes the `http.request.method`
+attribute recorded by the `latency` instrument from the `http` instrumentation
+library:
```
Comment (theletterf) [inline content/en/docs/instrumentation/go/manual.md:806]:
```suggestion
Here's how you create a view that removes the `http.request.method`
attribute recorded by the `latency` instrument from the `http` instrumentation
library:
```

File: content/en/docs/instrumentation/go/manual.md [hunk 5]
```diff
@@ -780,6 +780,64 @@ meterProvider := metric.NewMeterProvider(
...
+)
+```
+
+The `Name` field of criteria supports wildcard pattern matching. The wildcard
+`*` is recognized as matching zero or more characters, and `?` is recognized as
+matching exactly one character. For example, a pattern of `*` will match all
+instrument names.
+
+For example, here's how you create a view that sets unit to milliseconds for any
+instrument with a name suffix of `.ms`:
```
Comment (theletterf) [inline content/en/docs/instrumentation/go/manual.md:828]:
```suggestion
The `Name` field supports wildcard pattern matching. The `*` wildcard
is recognized as matching zero or more characters, and `?` is recognized as
matching exactly one character. For example, a pattern of `*` matches all
instrument names.

The following example show how you create a view that sets unit to
milliseconds for any instrument with a name suffix of `.ms`:
```

**Review by theletterf — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
thanks @pellared !

**Pull request merged**